src/extent.c

```diff
@@ -1930,6 +1967,11 @@ extent_merge_impl(tsdn_t *tsdn, arena_t *arena,
 		return true;
 	}
 
+	if (extent_dumpable_get(a) || extent_dumpable_get(b)) {
```

Comment:
Tricky part like this makes me wonder if we should go the other direction:
1) each arena keeps a special "no_dump" extent, which is the left over from ```grow_retained()```.
2) in grow_retained(), check the no_dump_extent first, and split / enable_dump as appropriate.

That way we sort of limit the scope of this dump / nodump thing in grow_retained itself. However the logic also feels more specialized.

The other thing is, this adds a branch on extent_alloc() which may translate to a tiny bit of cost. It won't be anything noticeable but makes this core dump thing a bit more irritating...

Thoughts?